magicblock-committor-service/src/intent_executor/intent_executor.rs

```diff
@@ -0,0 +1,247 @@
+        commit_signature: Signature,
+        persister: &Option<P>,
+    ) -> IntentExecutorResult<Signature> {
+        let prepared_message = self
+            .transaction_preparator
+            .prepare_finalize_tx(&self.authority, l1_message, persister)
+            .await
+            .map_err(Error::FailedFinalizePreparationError)?;
+
+        self.send_prepared_message(prepared_message).await.map_err(
+            |(err, finalize_signature)| Error::FailedToFinalizeError {
+                err,
+                commit_signature,
+                finalize_signature,
+            },
+        )
+    }
+
+    /// Shared helper for sending transactions
+    async fn send_prepared_message(
+        &self,
+        mut prepared_message: VersionedMessage,
+    ) -> IntentExecutorResult<Signature, (InternalError, Option<Signature>)>
+    {
+        let latest_blockhash = self
+            .rpc_client
+            .get_latest_blockhash()
+            .await
+            .map_err(|err| (err.into(), None))?;
+        match &mut prepared_message {
+            VersionedMessage::V0(value) => {
+                value.recent_blockhash = latest_blockhash;
+            }
+            VersionedMessage::Legacy(value) => {
+                warn!("TransactionPreparator v1 does not use Legacy message");
+                value.recent_blockhash = latest_blockhash;
+            }
+        };
+
+        let transaction =
+            VersionedTransaction::try_new(prepared_message, &[&self.authority])
+                .map_err(|err| (err.into(), None))?;
+        let result = self
+            .rpc_client
+            .send_transaction(
+                &transaction,
+                &MagicBlockSendTransactionConfig::ensure_committed(),
+            )
+            .await
+            .map_err(|err| {
+                let signature = err.signature();
+                (err.into(), signature)
+            })?;
+
+        Ok(result.into_signature())
+    }
+
+    fn persist_result<P: IntentPersister>(
+        persistor: &Option<P>,
+        result: &IntentExecutorResult<ExecutionOutput>,
+        message_id: u64,
+        pubkeys: &[Pubkey],
+    ) {
+        match result {
+            Ok(value) => {
+                let signatures = CommitStatusSignatures {
+                    process_signature: value.commit_signature,
+                    finalize_signature: Some(value.commit_signature)
+                };
+                let update_status = CommitStatus::Succeeded(signatures);
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::EmptyIntentError) => {
+                let update_status = CommitStatus::Failed;
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::FailedToFitError)) => {
```

Comment:
Import that Error (after giving it a proper name) and thus avoid these super long lines